util/src/main/java/org/datacommons/util/StatChecker.java

```diff
@@ -98,12 +97,21 @@ public StatChecker(LogWrapper logCtx, Set<String> samplePlaces, boolean verbose)
     }
   }
 
-  // Given a graph, extract time series info (about the chosen sample places) from the
-  // statVarObservation nodes and save it into seriesSummaryMap.
-  public void extractSeriesInfoFromGraph(McfGraph graph) {
+  // Given a graph, extract stat var info and time series info (about the chosen sample places) from
+  // the statVarObservation nodes.
+  public void extractStatsFromGraph(McfGraph graph) {
     for (Map.Entry<String, McfGraph.PropertyValues> node : graph.getNodesMap().entrySet()) {
-      if (isSvObWithNumberValue(node.getValue()) && shouldExtractSeriesInfo(node.getValue())) {
-        extractSeriesInfoFromNode(node.getValue());
+      if (isSvObWithNumberValue(node.getValue())) {
+        // We will extract basic stat var information from every StatVarObservation nodes
+        extractStatVarInfoFromNode(node.getValue());
+        // We will only extract series information from StatVarObservation nodes about sample places
+        if (shouldExtractSeriesInfo(node.getValue())) {
+          String placeDcid = McfUtil.getPropVal(node.getValue(), Vocabulary.OBSERVATION_ABOUT);
+          PlaceSeriesSummary placeSeriesSummary =
+              placeSeriesSummaryMap.getOrDefault(placeDcid, new PlaceSeriesSummary());
```

Comment:
I think this will allocate a new PlaceSeriesSummary() every time (given the heap pressure with large imports, this can matter a lot). Can we do: https://github.com/datacommonsorg/import/blob/7529566dcd5491ab376b3fadb193dd9c46d35db1/util/src/main/java/org/datacommons/util/StatChecker.java#L183

Reply:
done